lib/astutils.cpp

```diff
@@ -1084,6 +1084,35 @@ bool isEscapeFunction(const Token* ftok, const Library* library)
     return false;
 }
 
+static bool hasReturnFunction(const Token* tok, const Library* library, const Token** unknownFunc)
+{
+    if (!tok)
+        return false;
+    const Token* ftok = tok->str() == "(" ? tok->previous() : nullptr;
+    while (Token::simpleMatch(ftok, "("))
+        ftok = ftok->astOperand1();
+    if (ftok) {
+        const Function * function = ftok->function();
+        if (function) {
+            if (function->isEscapeFunction())
+                return true;
+            if (function->isAttributeNoreturn())
+                return true;
+        } else if (library && library->isnoreturn(ftok)) {
+            return true;
+        } else if (Token::Match(ftok, "exit|abort")) {
```

Comment:
Shouldn't the library contain `exit` and `abort` ?

Reply:
yes, this should be redundant to line 1101

Reply:
I think some of the tests dont use the library. I am not changing this part of the code just moving it.

Reply:
Actually this test fails:

https://github.com/danmar/cppcheck/blob/master/test/testvalueflow.cpp#L3841

Reply:
ok then the testing must be tweaked somehow. I guess adding "exit" is not very difficult.